src/flamenco/capture/solcap.md

```diff
@@ -0,0 +1,256 @@
+
+### Conceptual Layers
+
+Solcap uses a layered architecture with four distinct levels:
+
+```
+┌─────────────────────────────────────────────────────┐
+│  Layer 4: Message Payload                           │
+│  (Account updates, bank preimages, rewards, etc.)   │
+├─────────────────────────────────────────────────────┤
+│  Layer 3: Internal Chunk Header                     │
+│  (Message type identification + slot + txn_idx)     │
+├─────────────────────────────────────────────────────┤
+│  Layer 2: PCapNG Enhanced Packet Block              │
+│  (Packet framing, timestamps, length)               │
+├─────────────────────────────────────────────────────┤
+│  Layer 1: PCapNG File Structure                     │
+│  (Section Header + Interface Description)           │
+└─────────────────────────────────────────────────────┘
+```
+
+**Layer 1 - Base Layer (PCapNG File Structure):**
+- Provides file-level framing and metadata
+- Section Header Block (SHB) identifies the file format
+- Interface Description Block (IDB) defines capture characteristics
+
+**Layer 2 - Transport Layer (Enhanced Packet Blocks):**
+- Wraps each solcap message in a PCapNG Enhanced Packet Block (EPB)
+- Provides packet-level framing, timestamps, and length information
+- Enables compatibility with standard PCapNG tools
+
+**Layer 3 - Muxing Layer (Internal Chunk Header):**
+- Multiplexes different solcap message types within EPBs
+- Contains message type identifier (`block_type`)
+- Provides temporal context (`slot`) and ordering (`txn_idx`)
+- This is the primary demultiplexing layer for solcap
+
+**Layer 4 - Message Layer:**
+- Contains the actual solcap message payload
+- Format determined by `block_type` in Layer 3
+- Includes account updates, bank state, rewards, etc.
+
+---
+
+## Protocol Layers
+
+### Layer 1: Base Layer - PCapNG Framing
+
+The base layer provides file-level structure using the PCapNG format. This ensures compatibility with existing network capture tools while providing a robust container format.
+
+#### File Structure
+
+```
+[Section Header Block (SHB)]        ← File identification
+[Interface Description Block (IDB)] ← Capture interface metadata
+[Enhanced Packet Block (EPB) #1]    ← First message
+[Enhanced Packet Block (EPB) #2]    ← Second message
+...
+[Enhanced Packet Block (EPB) #N]    ← Nth message
+```
+
+The Section Header Block and Interface Description Block are written once at the beginning of the file. All subsequent messages are wrapped in Enhanced Packet Blocks.
+
+### Layer 2: Transport Layer - Enhanced Packet Blocks
+
+Each solcap message is encapsulated in a PCapNG Enhanced Packet Block (EPB). The EPB provides:
+- Block type identification
+- Total block length (for forward seeking)
+- Timestamp information
+- Packet length (captured vs original)
+- Redundant length footer (for backward seeking)
+
+These fields are not directly read in solcap files, and provide no use
+
+**Note**: Any Simple Packet Blocks are disregarded
```

Comment:
remove layer 1 and 2, these are already specified in the pcapng spec. You can just reference the pcapng spec